plugins/BTagAnalyzer.cc

```diff
@@ -1226,30 +1225,71 @@ void BTagAnalyzerT<IPTI,VTX>::analyze(const edm::Event& iEvent, const edm::Event
   //------------------------------------------------------
   if (use_ttbar_filter_) {
     edm::Handle<int> pIn;
-    iEvent.getByLabel(channel_, pIn);
+    iEvent.getByLabel(edm::InputTag("ttbarselectionproducer:topChannel"), pIn);
+    edm::Handle<int> triggerIn;
+    iEvent.getByLabel(edm::InputTag("ttbarselectionproducer:topTrigger"),triggerIn);
+    EventInfo.ttbar_trigWord=*triggerIn;
     EventInfo.ttbar_chan=*pIn;
-    edm::Handle<vector< double  >> pIn2;
-    iEvent.getByLabel(channel_, pIn2);
-    if (pIn2->size()==8) {
-      EventInfo.lepton1_pT=(*pIn2)[0];
-      EventInfo.lepton1_eta=(*pIn2)[1];
-      EventInfo.lepton1_phi=(*pIn2)[2];
-      EventInfo.lepton2_pT=(*pIn2)[3];
-      EventInfo.lepton2_eta=(*pIn2)[4];
-      EventInfo.lepton2_phi=(*pIn2)[5];
-      EventInfo.met=(*pIn2)[6];
-      EventInfo.mll=(*pIn2)[7];
-    }
-    else {
-      EventInfo.lepton1_pT=-1;
-      EventInfo.lepton1_eta=-1;
-      EventInfo.lepton1_phi=-1;
-      EventInfo.lepton2_pT=-1;
-      EventInfo.lepton2_eta=-1;
-      EventInfo.lepton2_phi=-1;
-      EventInfo.met=-1;
-      EventInfo.mll=-1;
-    }
+    int lctr(0);
+    edm::Handle<edm::View<pat::Electron> > selElectrons;
+    iEvent.getByLabel(ttbarproducer_,selElectrons);
+    for (size_t i = 0; i < selElectrons->size(); ++i)
+      {
+	const auto l = selElectrons->ptrAt(i);
+	EventInfo.ttbar_lpt[lctr]  = l->pt();
+	EventInfo.ttbar_leta[lctr] = l->eta();
+	EventInfo.ttbar_lphi[lctr] = l->phi();
+	EventInfo.ttbar_lm[lctr]   = 0;
+	EventInfo.ttbar_lch[lctr]  = l->charge();
+	EventInfo.ttbar_lid[lctr]  = 11;
+	EventInfo.ttbar_lgid[lctr] = l->genParticle() ? l->genParticle()->pdgId() : 0;
+	lctr++;
+      }
+
+    edm::Handle<edm::View<pat::Muon> > selMuons;
+    iEvent.getByLabel(ttbarproducer_,selMuons);
+    for (size_t i = 0; i < selMuons->size(); ++i)
+      {
+	const auto l = selMuons->ptrAt(i);
+	EventInfo.ttbar_lpt[lctr]  = l->pt();
+	EventInfo.ttbar_leta[lctr] = l->eta();
+	EventInfo.ttbar_lphi[lctr] = l->phi();
+	EventInfo.ttbar_lm[lctr]   = 0;
+	EventInfo.ttbar_lch[lctr]  = l->charge();
+	EventInfo.ttbar_lid[lctr]  = 11;
+	EventInfo.ttbar_lgid[lctr] = l->genParticle() ? l->genParticle()->pdgId() : 0;
+	lctr++;
+      }
+    EventInfo.ttbar_nl=lctr;
+
+    edm::Handle<edm::View<pat::MET> > selMETs;
+    iEvent.getByLabel(ttbarproducer_,selMETs);
+    EventInfo.ttbar_metpt=selMETs->ptrAt(0)->pt();
+    EventInfo.ttbar_metphi=selMETs->ptrAt(0)->phi();
+
+    edm::Handle< double > rhoH;
+    iEvent.getByLabel(rhoTag_,rhoH);
+    EventInfo.ttbar_rho = *rhoH;
+
+    //generator information
+    EventInfo.ttbar_nw=0;
+    if(!isData_)
+      {
+	try{
```

Comment:
It should be sufficient to check if a handle is valid by calling its `isValid()` method. So no need for the try-catch block and better leave the exception handling to CMSSW.